extensions/emmet/src/balance.ts

```diff
@@ -7,6 +7,10 @@ import * as vscode from 'vscode';
 import { HtmlNode } from 'EmmetNode';
 import { getNode, parseDocument, validate } from './util';
 
+let balanceStack: Array<vscode.Selection[]> = [];
```

Comment:
In the current code, the `balanceStack` is never reset. Take the case, where the user runs a series of balance out operations. Then they continue with some other work. After some time, they select some text and try to balance in. The current code will pop the stack and give wrong selections

Or take the case of a series of balance out, escape, then another series of balance out in a separate part of the file. Now when you balance in, the stack has selections from the first set which would be wrong